articles/ml_system_design/real_time_news_search_using_upstash_kafka_and_vector/upstash_ingest/upstash_ingest/consumer.py

```diff
@@ -0,0 +1,73 @@
+import json
+from typing import Iterable, List
+from kafka import KafkaConsumer
+
+from bytewax.inputs import DynamicSource, StatelessSourcePartition
+from models import CommonDocument
+from settings import settings
+
+from logger import get_logger
+
+logger = get_logger(__name__)
+
+
+class UpstashKafkaStreamSource(DynamicSource):
+    """Builds a KafkaStreamClient instance as a DynamicSource for Bytewax."""
+
+    def __init__(self):
+        self._consumer = KafkaStreamClient(
```

Comment:
Why haven't we used the Kafka connectors from Bytewax to keep the Bytewax code slimmer? https://docs.bytewax.io/stable/articles/concepts/kafka.html

Reply:
Hmm, interesting remark. 
Will give it a try, shouldn't take long at all.